### ISSUE TITLE
confirmation for the sender for over the contact form sent e-mail

### ISSUE THREAD

**Issue by auge8472** [ilosuna/mylittleforum#238]
In ancient times the sender of an e-mail, sent over the contact form, got an confirmation mail. Because the sender was able to change the address to which the confirmation was sent, it was a possible source of spam. Because of questions it's appropriate to think about a reimplementation of this feature without the risk of being a spam source (as far as possible).

I for myself would give it a go.

Therefore see also these threads in the project forum:

- [actual discussion](http://mylittleforum.net/forum/index.php?id=10153),  asking for confirmation mail
- [old thread](http://mylittleforum.net/forum/index.php?id=3420), asking also for confirmation mail (from 2008)
- [old thread](http://mylittleforum.net/forum/index.php?id=2485) stating the function as possible spam source (also from 2008)

**Comment by auge8472** [comment]
I recapitulated the most recent discussion about confirmation e-mails. @derletztekick [stated the conclusion](http://mylittleforum.net/forum/index.php?id=10158) of the old discussion not to send a confirmation because [it was a possible source of spam in it's implementation at that time](http://mylittleforum.net/forum/index.php?id=2485). Is it wishful to send a copy of an e-mail (under the premise that the sender acts as a registered user and can not change the e-mail-address)? Or would it be a renewed security problem or a hidden spam source because of incomplete [considerations](http://mylittleforum.net/forum/index.php?id=10160)?

**Comment by derletztekick** [comment]
I cannot found any weak points in your suggestion. I also confirm the first three points of your list:

- Not logged in users should not be able to receive a copy of the e-mail.
- Logged in users should be able to receive a copy of the e-mail.
- The ability to change the senders e-mail-address is a no go.

The last one, well, I think we should(?) add a checkbox "Sent me a copy of this message, too".
- IMHO the copy of the e-mail should not be mandatory.

Your pseudo code contains a repetition (is this the right word in this context?):

```
if (recipient-is-valid) {
      // send e-mail to the recipient
 }
```

I would change this part to:
```
if ((send-with-copy OR send-without-copy) AND recipient-is-valid) {
  // send e-mail to the recipient (always!)

  // mail form was sent by a visitor or registered user
  // form was sent by a registered user
  if (user-is-registered AND send-with-copy AND sender-is-valid) {
      // send additional e-mail to the sender
  }
}
```
Basically, I support your investigation.

**Comment by auge8472** [comment]
> Basically, I support your investigation.

I began to study the `contact.inc.php` and reformatted the script for better readability in a first step. I think, most necessary checks (valid sender, valid recipient, handling of user-ids or entry-ids instead of e-mail-addresses) are in general present. Good precondition for the start.

**Comment by auge8472** [comment]
This feature depends on #348.

**Comment by loesler** [comment]
A copy of the mail is send to the sender, if sender is a reg. user. Done by https://github.com/ilosuna/mylittleforum/pull/501
